src/message-reader.cpp

```diff
@@ -34,32 +36,43 @@ void MessageReader::_clearUntilStart(bool forceInitialShift)
     }
 }
 
+void MessageReader::_printBuffer(){
+    for (int i = 0; i < this->buffer.size(); i++)
+    {
+        printf("%c", this->buffer[i]);
```

Comment:
I'm not sure if this really works in the arduino.  you might want to pass in a logger as done in the system implementations.

Reply:
Changed it to use an injected ILogger instead. I also implemented the methods in the test version.